sdk/python/kubeflow/training/utils/utils.py

```diff
@@ -323,13 +333,13 @@ def get_pytorchjob_template(
         spec=models.KubeflowOrgV1PyTorchJobSpec(
             run_policy=models.KubeflowOrgV1RunPolicy(clean_pod_policy=None),
             pytorch_replica_specs={},
+            elastic_policy=elastic_policy,
```

Comment:
should we check if elastic policy is not an empty dict, else default env variables will get appended
https://github.com/kubeflow/training-operator/blob/0b6a30cd348e101506b53a1a176e4a7aec6e9f09/pkg/controller.v1/pytorch/envvar.go#L115

Reply:
Actually, it elastic_policy is None, Python doesn't assign value to the PyTorchJob spec, and we don't set default values.
If user accidentally set `elastic_policy={}`, our controller will fail with invalid spec error:
```
E0118 15:47:43.737955       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 398 [running]:
```

Reply:
As you can see, elastic_policy has this type: `KubeflowOrgV1ElasticPolicy`, so user should set the appropriate instance value similar to other paras (e.g. worker_pod_template_spec).
Right now, we don't even use elastic_policy in our public APIs: https://github.com/kubeflow/training-operator/blob/174b050cce88342b29bf3e098a67e5afa9d3fb9a/sdk/python/kubeflow/training/api/training_client.py#L237.